java/src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -338,6 +341,10 @@ public Integer call() {
             logger.info("-> Validation will use " + threads + " threads.");
             validator.setThreadCount(threads);
         }
+        if (debugOutput) {
+            logger.info("-> Multi-threaded validation debugging output enabled.");
+            ((Logger) (org.slf4j.LoggerFactory.getLogger(ThreadedValidationEngine.class))).setLevel(Level.DEBUG);;
+        }
```

Comment:
As the ThreadedValidationEngine is abstracted from what the CLI is aware of, I would like to see something in the ValidateAIF object that would enable debugging. It would be ideal if there was some way to combine this and the progress monitor into a single idea for ValidateAIF. To just get output, I'm fine with the current implementation, but if we're going to commit this, we should provide controls for this in the ValidateAIF object. Let me know if you want to discuss.

Reply:
@craigwarsaw has said that he wanted this change committed as an undocumented command line option.  I agree that it would be nice to have the CLI class have no knowledge of the `ThreadedValidationEngine`.  Unfortunately it already does in how it refers to `ThreadedValidationEngine.SH_ABORTED` to see if the validation was aborted (although I see now that there's a `TODO` note to "come up with better property").

It just seems a little bit...misleading? to add a debugging parameter/flag/etc at the ValidateAIF if we have no intention of implementing anything for it other than in the `ThreadedValidationEngine`.  I also thought about having `--debug` require the `-t` flag, but that seemed like unnecessary CLI wrangling for a developer-only tool...but I could be convinced otherwise.

Happy to discuss any of this further.

Reply:
Perhaps I should also change the flag to `--debugMT`.